### PR TITLE
Configurations generator remove istio policy

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Changes proposed in this pull request:

- Remove Istio Authentication Policy from configurations-generator
- Bump configurations-generator version to the one with support for JWT token authentication
- Use configurations-generator built-in JWT token authentication mechanism

See also #2525

### CODE REVIEW

File: resources/dex/templates/dex-config-map.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ data:
     issuer: https://dex.{{ .Values.global.domainName }}
     # extending it to 15days because apparently Istio does not handle ID token validation properly
     expiry:
-      signingKeys: "360h"
+      signingKeys: "15m"
     storage:
```
Comment (piotrmsc) [inline resources/dex/templates/dex-config-map.yaml]: this is already covered in https://github.com/kyma-project/kyma/pull/2736
where btw keys should expire after 1 month :)

Comment (Tomasz-Smelcerz-SAP) [reply resources/dex/templates/dex-config-map.yaml]: Yes, this is for testing purposes only. Will be removed before merge.

**Comment by Tomasz-Smelcerz-SAP** [comment]
Note for reviewer: The following procedure can be used to reproduce the problem on master and/or verify that the problem is fixed on the P/R branch. 

How to reproduce the problem: (Use Kyma installation that DOES NOT contain fixes from this PR):

- Set up dex expiration key time to about 15 mins (resources/dex/templates/dex-config-map.yaml)
- Launch Kyma locally on Minikube
- Once installed, log in to console.kyma.local
- Verify you can download the configuration. Stay logged in!
- Wait until dex rotates the keys (take a look at dex logs)
- Verify you can still download the configuration (without logging out!) - console app still has an old token, should work.
- Logout and log-in again. Console app has a new token, but the Istio authentication policy can't verify it because it doesn't have new dex keys (it keeps old one in cache)
- Verify you can't download configuration anymore.

**Pull request merged**